xl-objects.go

```diff
@@ -470,7 +473,17 @@ func (xl xlObjects) deleteMultipartObject(bucket, object string) error {
 		go func(index int, part MultipartPartInfo) {
 			defer wg.Done()
 			partFileName := partNumToPartFileName(part.PartNumber)
-			errs[index] = xl.storage.DeleteFile(bucket, pathJoin(object, partFileName))
+			tempObj := path.Join(tmpMetaPrefix, bucket, pathJoin(tempObjID.String()+"."+object, partFileName))
+			err = xl.storage.RenameFile(bucket, pathJoin(object, partFileName), minioMetaBucket, tempObj)
+			if err != nil {
+				// Ignore if object doesn't exist.
+				if err != errFileNotFound {
+					errs[index] = err
+					return
+				}
+			}
+
+			errs[index] = xl.storage.DeleteFile(minioMetaBucket, tempObj)
```

Comment:
the intention of rename here is so that we retain the object atleast in `tmp` worst case we can recover from here, so deleting it right after rename will not serve the purpose